docs/architecture/018-ega-integration.md

```diff
@@ -0,0 +1,39 @@
+# 018: EGA Integration
+
+Authors: @Macroz
+
+## Situation
+
+REMS instances, such as the THL Biobank, wish to synchronize (push) access rights (entitlements) to The European Genome-phenome Archive (EGA).
+
+We have an existing entitlement post function, but that does not support the kind of features that EGA requires (like API-key per handler).
+
+REMS also has a permissions API that supports GA4GH style permission visas. This is for the cases where the entitlements are fetched from REMS.
+
+EGA data download tools check the access rights from EGA servers so we must build something new to push the entitlements there.
+
+The EGA Permissions API is [documented here](https://docs.google.com/document/d/1FTzUYAfV5d2a0zoDkbY9Iy_L5NbSAnHeWnmY2NIrY8M/edit?usp=drivesdk).
+
+## Solution
+
+To enable pushing permissions, we have implemented a new v2 entitlement push. It supports multiple simultaneous integrations as well as types, 
+of which ˋ:egaˋ is the only one implemented so far.
+
+The push is done by the same entitlement background task as before.
+
+For each handler there must be an EGA API-key stored in the user secrets, otherwise the entitlement can't be pushed.
```

Comment:
… otherwise entitlements can't be pushed.

Reply:
I also added a note that it concerns the entitlements approved by that handler.